clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ClusteredStateRepository.java

```diff
@@ -28,17 +30,19 @@
 class ClusteredStateRepository implements StateRepository {
 
   private final ClusteringService.ClusteredCacheIdentifier clusterCacheIdentifier;
+  private final EhcacheClientEntity clientEntity;
   private final String composedId;
-  private final DefaultClusteringService defaultClusteringService;
+  private final ConcurrentMap<String, ConcurrentClusteredMap> knownMaps = new ConcurrentHashMap<String, ConcurrentClusteredMap>();
 
-  ClusteredStateRepository(ClusteringService.ClusteredCacheIdentifier clusterCacheIdentifier, String id, DefaultClusteringService defaultClusteringService) {
+  ClusteredStateRepository(ClusteringService.ClusteredCacheIdentifier clusterCacheIdentifier, String id, EhcacheClientEntity clientEntity) {
     this.clusterCacheIdentifier = clusterCacheIdentifier;
     this.composedId = clusterCacheIdentifier.getId() + "-" + id;
-    this.defaultClusteringService = defaultClusteringService;
+    this.clientEntity = clientEntity;
   }
 
   @Override
   public <K extends Serializable, V extends Serializable> ConcurrentMap<K, V> getPersistentConcurrentMap(String name, Class<K> keyClass, Class<V> valueClass) {
-    return defaultClusteringService.getConcurrentMap(clusterCacheIdentifier, composedId + "-" + name, keyClass, valueClass);
+    knownMaps.putIfAbsent(name, new ConcurrentClusteredMap<K, V>(clusterCacheIdentifier.getId(), composedId + "-" + name, clientEntity));
+    return knownMaps.get(name);
```

Comment:
With the delayed creation logic as in the following commit, this should be fine, right?